packages/db/src/query/optimizer.ts

```diff
@@ -424,14 +432,44 @@ function isRedundantSubquery(query: QueryIR): boolean {
  * ```
  */
 function splitAndClauses(
+  whereClauses: Array<Where>
+): Array<BasicExpression<boolean>> {
+  const result: Array<BasicExpression<boolean>> = []
+
+  for (const whereClause of whereClauses) {
+    // Skip residual WHERE clauses to prevent them from being pushed down again
+    if (isResidualWhere(whereClause)) {
+      const clause = getWhereExpression(whereClause)
```

Comment:
We can move this constant to L440 (i.e. before the if statement) and then remove the one on L447